grafana/grafana
# Elasticsearch: Decouple frontend dependencies from core

This PR is the last step to decouple frontend core dependecies for Elasticsearch data source. This PR does couple of thing:
- Add Elasticsearch to `.eslintrc` to prevent importing from core
- Creates `flattenObject` and  `variableRegex`  which is a copy from core
- A lot of changing of absolute path to relative path 
- Instead of using `selectOptionInTest`, use `select` directly
- Creates own `reducerTester` instead of `test/core/redux/reducerTester`
- Use `DataSourcePicker` from `@grafana/runtime`

Maybe other things.

## Testing
Make sure that you run the code and try to:
1. Update config page with Derived fields - data source picker works
1. Run some queries and make sure no errors

Part of https://github.com/grafana/grafana/issues/72632
